src/main/java/gift/ProductController.java

```diff
@@ -0,0 +1,72 @@
+package gift;
+
+import jakarta.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+  private final ProductDao productDao;
+
+  @Autowired
+  public ProductController(ProductDao productDao) {
+    this.productDao = productDao;
+  }
+
+
+  @GetMapping
+  public List<Product> getAllProducts() {
+    List<Product> productDTO = new ArrayList<Product>();
+    productDTO=productDao.selectAllProducts();
```

Comment:
```suggestion
    List<Product> productDtos = productDao.selectAllProducts();
```
이런식으로 카멜케이스로 나타내면 좋을거 같습니다~ 아울러 재할당 할 이유가 없어보여요!
마지막으로 Collection은 복수형으로 변수를 지어서 나타내는 편입니다!